### PR TITLE
kokkos-devel: new port to track upstream, with support for 32-bit

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

This is based on `develop` branch of upstream.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6.8 Server
Xcode 3.2.6

macOS 10A190
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by barracuda156** [comment]
@MarcusCalhoun-Lopez Hope you don’t mind me adding you. Linking with Clang I will fix now.

File: devel/kokkos-devel/Portfile [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem                  1.0
+PortGroup                   github 1.0
+PortGroup                   cmake 1.1
+
+github.setup                kokkos kokkos 5b1a0e36a84994244f24efe3a199b324be455b05
+name                        kokkos-devel
```
Comment (reneeotten) [inline devel/kokkos-devel/Portfile:8]: can this not be added as a subport of kokkos? I assume many parts of the Portfile will be identical

Comment (barracuda156) [reply devel/kokkos-devel/Portfile:8]: I guess it can, if that is preferred. `kokkos` is not my port, so I went with a less intrusive option.

Comment (barracuda156) [reply devel/kokkos-devel/Portfile]: @reneeotten Done.

**Comment by barracuda156** [comment]
GitHub again broken checksums… Gonna fix today, didn’t notice earlier.

**Comment by barracuda156** [comment]
@reneeotten Could you please take a look?

**Review by reneeotten — CHANGES_REQUESTED** [review]
this is not how we typically do subports. You keep everything in the "main" Portfile as it was and just create a subport where you repeat the github.setup line and dk whatever else is specific to that port.

File: devel/kokkos/Portfile [hunk 2]
```diff
@@ -29,8 +45,21 @@ configure.args-append       -DBUILD_SHARED_LIBS=ON \
...
+if {[string match *gcc* ${configure.compiler}]} {
+    configure.ldflags-append \
+                            -latomic
+}
+
+if {[string match *clang* ${configure.compiler}]} {
+    configure.ldflags-append \
+                            -L${prefix}/lib/libomp \
+                            -lomp
+}
```
Comment (reneeotten) [inline devel/kokkos/Portfile:57]: an this not be done automagically by setting "compiler.openmp_version" instead?

Comment (barracuda156) [reply devel/kokkos/Portfile:57]: I will check. I have seen other ports adding this manually, see. `libfaiss`.

**Comment by barracuda156** [comment]
I will update the PR with the version merged to upstream in https://github.com/kokkos/kokkos/commit/e5490e1e11ece175ec16588c5a3062ece2fe1b33

File: devel/kokkos/Portfile (hunk 2 above)
Comment (barracuda156) [reply devel/kokkos/Portfile]: Perhaps not needed, let us see if everything builds. Rebased now.

Comment (barracuda156) [reply devel/kokkos/Portfile]: @reneeotten Well, no, at least for `kokkos-devel` without explicit linking it fails:
```
Undefined symbols for architecture x86_64:
    "___kmpc_for_static_fini", referenced from:
        _.omp_outlined. in Kokkos_HostSpace_deepcopy.cpp.o
        _.omp_outlined..7 in Kokkos_HostSpace_deepcopy.cpp.o
        _.omp_outlined..8 in Kokkos_HostSpace_deepcopy.cpp.o
    "___kmpc_for_static_init_8u", referenced from:
        _.omp_outlined. in Kokkos_HostSpace_deepcopy.cpp.o
        _.omp_outlined..7 in Kokkos_HostSpace_deepcopy.cpp.o
        _.omp_outlined..8 in Kokkos_HostSpace_deepcopy.cpp.o
```

**Comment by barracuda156** [comment]
@reneeotten Now should be ready to merge.

**Comment by barracuda156** [comment]
@reneeotten Bump, let’s merge this finally, it is already in upstream, patches dropped.

**Comment by reneeotten** [comment]
As I said before this is not how subports are typically implemented. Add the code into the subport definition that saves "if-statements". See, for example, the "pymol" Portfile among many others.

**Comment by barracuda156** [comment]
> As I said before this is not how subports are typically implemented. Add the code into the subport definition that saves "if-statements". See, for example, the "pymol" Portfile among many others. 

@reneeotten Well, I thought you didn’t like it being in a separate port (for which there are multiple existing examples too, BTW), so moved it into the same portfile. The style was used which is also used in a number of other ports, I did not invent it. But okay, I can redo it to match the style of `pymol`.

**Comment by barracuda156** [comment]
@reneeotten Is it okay now?

**Comment by barracuda156** [comment]
@reneeotten Thank you!

**Pull request merged**